st2actions/st2actions/handlers/mistral.py

```diff
@@ -40,17 +40,30 @@ class MistralCallbackHandler(handlers.ActionExecutionCallbackHandler):
 
     @staticmethod
     def callback(url, context, status, result):
+        if status not in [action.ACTIONEXEC_STATUS_SUCCEEDED, action.ACTIONEXEC_STATUS_FAILED]:
+            return
+
         try:
             method = 'PUT'
+            headers = {'content-type': 'application/json'}
+
             if isinstance(result, basestring) and len(result) > 0 and result[0] in ['{', '[']:
                 value = ast.literal_eval(result)
                 if type(value) in [dict, list]:
                     result = value
-            output = json.dumps(result) if type(result) in [dict, list] else str(result)
+
+            if type(result) in [dict, list]:
+                # Remove the list of tasks created by the results
+                # tracker before sending the output.
+                result.pop('tasks', None)
```

Comment:
If result is a list this will throw since pop in list only takes one argument (index).